pkg/controller/endpointslice/utils.go

```diff
@@ -158,8 +158,8 @@ func newEndpointSlice(service *corev1.Service, endpointMeta *endpointMeta) *disc
 	ownerRef := metav1.NewControllerRef(service, gvk)
 	return &discovery.EndpointSlice{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels:          map[string]string{serviceNameLabel: service.Name},
-			GenerateName:    fmt.Sprintf("%s.", service.Name),
+			Labels:          map[string]string{discovery.LabelServiceName: service.Name},
+			GenerateName:    fmt.Sprintf("%s-", service.Name),
```

Comment:
if the service is the maximum allowed length, does this produce an invalid endpointslice name?

Reply:
Is there a utility associated with GenerateName? It seems like this should be a pretty generic problem associated with any use

Reply:
Good catch, I've updated this to use the pattern used with pods. A shared helper func would make sense here, but I can't find one.